src/api/app/policies/attrib_policy.rb

```diff
@@ -39,4 +20,32 @@ def update?
   def destroy?
     create?
   end
+
+  private
+
+  # Parses AttribTypeModifiableBy and AttribNamespaceModifiableBy association
+  # collections and checks the users permissions.
+  #
+  # Permission check is based on:
+  #   - User
+  #   - Group association of user
+  #   - Role association of user (Only for AttribTypeModifiableBy)
+  def has_permissions_for_modifiables?(attrib_modifiable_bies)
+    attrib_modifiable_bies.to_a.any? do |rule|
+      rule.user == @user ||
+      @user.is_in_group?(rule.group) ||
+        (rule.try(:role) && @user.has_local_role?(rule.role, @record.container))
+    end
+  end
```

Comment:
:+1:

And I think I found a nicer way to further simplify:  f4100bd1c4fa004afa09c93cd910ff51edfb32bd